### PR TITLE
Port upstream xz 5.4.4

### PR DESCRIPTION
**Pull request by andreittr**
Files under include/ are generated by the upstream build and adapted for Unikraft.

### CODE REVIEW

File: Config.uk
Comment (StefanJum) [inline Config.uk]: I kind of do not like this, since we allow other libcs by using `HAVE_LIBC`, but enforce the use of `musl` with `LIBMUSL_THREAD`.
I don't really see a way out though, so I wouldn't block on this unless you can see a better way to do it.

**Review by StefanJum — COMMENTED** [review]
It works fine, just one comment.

Also this does not work for arm64, I guess that's known, right?

File: Config.uk
Comment (andreittr) [reply Config.uk]: We could use `HAVE_PTHREAD`, but that can't account for how much pthread support a lib has (see longer response https://github.com/unikraft/lib-littlecms/pull/1#discussion_r1342675762). In this case I'm also tempted to remove this option until (1) SMP support lands, so it actually makes sense and (2) we sort out the libc/pthread situation.

**Comment by andreittr** [comment]
Force-push: Removed problematic multi-thread option, will revisit at later date when needed.
(Option ifdefs are left in place, to ease enabling once we do get back to it)

**Review by StefanJum — APPROVED** [review]
All good.
Reviewed-by: Stefan Jumarea <stefanjumarea02@gmail.com>

**Review by mariasfiraiala — APPROVED** [review]
All good on my side, as well! Thanks!

Reviewed-by: Maria Sfiraiala <maria.sfiraiala@gmail.com>

**Review by razvand — APPROVED** [review]
Approved-by: Razvan Deaconescu <razvand@unikraft.io>

**Pull request closed without merging**
